packages/yswaps/scripts/execute-mainnet-pending-trades.ts

```diff
@@ -0,0 +1,108 @@
+import { TradeFactory } from '@typechained';
+import zrx from './libraries/zrx';
+import { UNISWAP_V2_FACTORY, UNISWAP_V2_ROUTER, WETH } from '@deploy/mainnet-swappers/uniswap_v2';
+
+const DELAY = moment.duration('3', 'minutes').as('milliseconds');
+const FLASHBOT_PROVIDER = new ethers.providers.JsonRpcProvider('https://rpc.flashbots.net');
+const SLIPPAGE_PERCENTAGE = 3;
+
+type PendingTrade = [BigNumber, string, string, string, BigNumber, BigNumber] & {
+  _id: BigNumber;
+  _strategy: string;
+  _tokenIn: string;
+  _tokenOut: string;
+  _amountIn: BigNumber;
+  _deadline: BigNumber;
+};
+
+type TradeSetup = {
+  swapper: string;
+  data: string;
+  minAmountOut: BigNumber | undefined;
+};
+
+async function main() {
+  const chainId = await getChainId();
+  console.log('Chain ID:', chainId);
+  let [flashBotSigner] = await ethers.getSigners();
+  flashBotSigner = flashBotSigner.connect(FLASHBOT_PROVIDER);
+
+  const tradeFactory = await ethers.getContract<TradeFactory>('TradeFactory');
+  const pendingTradesIds = await tradeFactory['pendingTradesIds()']();
+  const pendingTrades: PendingTrade[] = [];
+  const tradesSetup: TradeSetup[] = [];
+
+  for (const id of pendingTradesIds) {
+    pendingTrades.push(await tradeFactory.pendingTradesById(id));
+  }
+
+  for (const pendingTrade of pendingTrades) {
+    if (pendingTrade._deadline.lt(moment().unix())) {
+      console.log(`Expiring trade ${pendingTrade._id.toString()}`);
+      await tradeFactory.expire(pendingTrade._id);
+      continue;
+    }
+
+    const { data: uniswapV2Data, minAmountOut: uniswapV2MinAmountOut } = await uniswap.getBestPathEncoded({
+      tokenIn: pendingTrade._tokenIn,
+      tokenOut: pendingTrade._tokenOut,
+      amountIn: pendingTrade._amountIn,
+      uniswapV2Router: UNISWAP_V2_ROUTER,
+      uniswapV2Factory: UNISWAP_V2_FACTORY,
+      hopTokensToTest: [WETH],
+      slippage: SLIPPAGE_PERCENTAGE,
+    });
+
+    tradesSetup.push({
+      swapper: (await ethers.getContract('AsyncUniswapV2')).address,
+      data: uniswapV2Data,
+      minAmountOut: uniswapV2MinAmountOut,
+    });
+
+    console.log('uniswap v2:', uniswapV2MinAmountOut, uniswapV2Data);
+
+    const { data: zrxData, minAmountOut: zrxMinAmountOut } = await zrx.quote({
+      chainId: Number(chainId),
+      sellToken: pendingTrade._tokenIn,
+      buyToken: pendingTrade._tokenOut,
+      sellAmount: pendingTrade._amountIn,
+      slippagePercentage: SLIPPAGE_PERCENTAGE / 100,
+    });
+
+    tradesSetup.push({
+      swapper: (await ethers.getContract('ZRX')).address,
+      data: zrxData,
+      minAmountOut: zrxMinAmountOut,
+    });
+
+    console.log('zrx:', zrxMinAmountOut, zrxData);
+
+    let bestSetup: TradeSetup = tradesSetup[0];
+
+    for (let i = 1; i < tradesSetup.length; i++) {
+      if (tradesSetup[i].minAmountOut!.gt(bestSetup.minAmountOut!)) {
+        bestSetup = tradesSetup[i];
+      }
+    }
+
+    const tx = await tradeFactory
+      .connect(flashBotSigner)
```

Comment:
@lbertenasco have you ever tested this approach ?

Reply:
haven't tested it, but you should use block protection contract.
https://etherscan.io/address/0xcc268041259904bb6ae2c84f9db2d976bceb43e5

Reply:
you'll have more control by manually submitting to flashbots, rpc is a bit buggy specially on tx replacement.